### ISSUE TITLE
Binding to the service in the play store failed . This is due to having an old play store version installed on the device

### ISSUE THREAD

**Issue by amlan007** [microg/FakeStore#36]
Binding to the service in the play store failed . This is due to having an old play store version 

Is this fake store issue or gms core issue

**Comment by ildar** [comment]
1st, you need to provide much more details.
http://www.catb.org/~esr/faqs/smart-questions.html
2nd, please ask at MicroG support. This small app is just a stub.

**Comment by paolo-caroni** [comment]
If this is a Play Integrity API check problem it seems to be a fakestore issue, see [error codes](https://developer.android.com/google/play/integrity/error-codes).
But honestly I'm not sure, I know that all app that use Play Integrity give me this error.
I don't know if gmscore implement such API, since is not listed on [Implementation Status](https://github.com/microg/GmsCore/wiki/Implementation-Status).

**Comment by paolo-caroni** [comment]
>[Play Integrity API Checker](https://play.google.com/store/apps/details?id=gr.nikolasspyr.integritycheck) report success on my phone.
>
>It is possible that the problem is not microG.
I'm not completely sure but I think Play Integrity API is just SafeNet with some additional checks and maybe a different interface but the base checks are the same.

_Originally posted by @ale5000-git in https://github.com/microg/GmsCore/issues/1771#issuecomment-1551883186_


For me is not yet very clear, but implement the api on gmscore and link with fakestore, or implement on fekestore is probably indifferent.
The important issue is implement it, as asked on more and more issues.

**Comment by redacted-text** [comment]
> > [Play Integrity API Checker](https://play.google.com/store/apps/details?id=gr.nikolasspyr.integritycheck) report success on my phone.
> > It is possible that the problem is not microG.
> > I'm not completely sure but I think Play Integrity API is just SafeNet with some additional checks and maybe a different interface but the base checks are the same.
> 
> _Originally posted by @ale5000-git in [microg/GmsCore#1771 (comment)](https://github.com/microg/GmsCore/issues/1771#issuecomment-1551883186)_
> 
> For me is not yet very clear, but implement the api on gmscore and link with fakestore, or implement on fekestore is probably indifferent. The important issue is implement it, as asked on more and more issues.

```java
.setPackage("com.android.vending");
```
Code from Play Integrity library.
It should be implemented in fake store.

**Comment by Uldiniad** [comment]
@mar-v-in 

One of the functions that needs implementing https://developer.android.com/google/play/integrity/reference/com/google/android/play/core/integrity/IntegrityManager.html#requestIntegrityToken(com.google.android.play.core.integrity.IntegrityTokenRequest)

Log from successful attempt on a GMS device
```2023-05-26 13:14:23.042 25976-25976 PlayCore                gr.nikolasspyr.integritycheck        I  UID: [10226]  PID: [25976] IntegrityService : requestIntegrityToken(IntegrityTokenRequest{nonce=6q3Lt9KRg6mi9TpwtsNa8zW3gGvXGYRYD0mo0bwvmBrrlHpY9d, cloudProjectNumber=null})
2023-05-26 13:14:23.042 25976-26009 PlayCore                gr.nikolasspyr.integritycheck        I  UID: [10226]  PID: [25976] IntegrityService : Initiate binding to the service.
2023-05-26 13:14:23.065 25976-25976 PlayCore                gr.nikolasspyr.integritycheck        I  UID: [10226]  PID: [25976] IntegrityService : ServiceConnectionImpl.onServiceConnected(ComponentInfo{com.android.vending/com.google.android.finsky.integrityservice.IntegrityService})
2023-05-26 13:14:23.065 25976-26009 PlayCore                gr.nikolasspyr.integritycheck        I  UID: [10226]  PID: [25976] IntegrityService : linkToDeath
2023-05-26 13:14:25.952  4742-5927  Finsky                  com.android.vending                  I  [54] hjt.b(30): requestIntegrityToken() finished for gr.nikolasspyr.integritycheck.
2023-05-26 13:14:25.956 25976-26900 PlayCore                gr.nikolasspyr.integritycheck        I  UID: [10226]  PID: [25976] OnRequestIntegrityTokenCallback : onRequestIntegrityToken
2023-05-26 13:14:25.957 25976-26009 PlayCore                gr.nikolasspyr.integritycheck        I  UID: [10226]  PID: [25976] IntegrityService : Unbind from service.```

**Comment by redacted-text** [comment]
I believe all play store does it's just says "yep, this app is licensed", and then forwards request to droidguard.

**Comment by Uldiniad** [comment]
> @mar-v-in, I can try to implement this feature, I reversed it a lot already. Is it welcome to create a PR? Asking cause It's probably few days of work I used microG a lot, I believe it's time to pay back :)

Since this is an open-source project, yes, PRs are welcome

**Comment by redacted-text** [comment]
@Uldiniad you will work on this?

**Comment by Uldiniad** [comment]
> weekend

No, I just implemented the stub as a starting point. Feel free to take it on from here

**Comment by paolo-caroni** [comment]
> I believe all play store does it's just says "yep, this app is licensed", and then forwards request to droidguard.

It seems to me to be a bit more complicated, since is a [signed response](https://developer.android.com/google/play/integrity/verdict).
I wish you good luck on reverse engine it.

**Comment by bbaster** [comment]
@redacted-text Hello, is there any new progress on this?

**Comment by paolo-caroni** [comment]
@bbaster I have contacted him on telegram, it seems that play integrity continue to change internally day to day... He is waiting that google stabilize it. Then will be a pull request.
